modules/shared-utils/src/cljdoc/util.clj

```diff
@@ -36,6 +37,28 @@
   {:pre [(some? project) (string? version)]}
   (str "cljdoc-edn/" (group-id project) "/" (artifact-id project) "/" version "/cljdoc.edn"))
 
+(defn- wrap-regex-tag [potential-regex]
+  (if (instance? java.util.regex.Pattern potential-regex)
+    (tagged-literal 'regex (str potential-regex))
+    potential-regex))
+
+(defn- wrap-all-regex-tags [structure]
+  (clojure.walk/postwalk wrap-regex-tag structure))
+
+(defn- unwrap-regex-tag [regex-tagged-string]
+  (re-pattern regex-tagged-string))
+
+(defn serialize-cljdoc-edn [analyze-result]
+  ;; the analyzed structure can contain regex #"..." (e.g. in :arglists)
+  ;; and they can't be read in again using edn/read-string
+  ;; so there are changed to #regex"..." and read in with a custom reader
+  (-> analyze-result
+      (wrap-all-regex-tags)
+      (pr-str)))
+
+(defn read-cljdoc-edn [file]
+  (edn/read-string {:readers {'regex unwrap-regex-tag}} (slurp file)))
```

Comment:
The redirection here makes the code slightly harder to read IMHO. What do you think about?
```suggestion
  (edn/read-string {:readers {'regex re-pattern}} (slurp file)))
```

Reply:
IMHO having this indirection only makes sense when the `wrap-regex-tag` is it's own function, as it makes it clear that they are opposites.